csharp/src/Ice/LocatorInfo.cs

```diff
@@ -354,14 +432,13 @@ async Task<IReadOnlyList<Endpoint>> PerformGetAdapterEndpointsAsync(Reference re
             {
                 try
                 {
-                    // TODO: Fix FindObjectById to return non-null proxy
                     IObjectPrx? proxy = await Locator.FindObjectByIdAsync(
                         reference.Identity,
                         cancel: CancellationToken.None).ConfigureAwait(false);
                     if (proxy != null && !proxy.IceReference.IsWellKnown)
```

Comment:
it will be good to add a comment here about why we do have to check "IsWellKnown"

Reply:
I think the logic is FindObjectById (i.e. "resolve this well-known proxy") is not allowed to return another well-known proxy. Hence if it does, we return null to the client code, just as if we didn't find anything.